internal/policies/mount/internal_test.go

```diff
@@ -0,0 +1,70 @@
+package mount
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var update bool
+
+func TestWriteMountsFile(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		path        string
+		nEntries    int
+		nErrored    int
+		nValues     int
+		nDuplicated int
+		separators  []string
+
+		wantErr bool
+	}{
+		"write mounts file with multiple entries":                          {nEntries: 5, nValues: 1},
+		"write mounts file with values separated by ','":                   {nEntries: 1, nValues: 5, separators: []string{","}},
+		"write mounts file with values separated by '\n'":                  {nEntries: 1, nValues: 5},
+		"write mounts file with values separated by ',' and '\n'":          {nEntries: 1, nValues: 5, separators: []string{",", "\n"}},
+		"write mounts file with values deduplicated from values":           {nEntries: 1, nValues: 5, nDuplicated: 3},
+		"write mounts file with values deduplicated from multiple entries": {nEntries: 3, nValues: 5, nDuplicated: 3},
+
+		"errored entries should not be added": {nEntries: 5, nErrored: 3, nValues: 2},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			mountsPath := filepath.Join(t.TempDir(), "mounts")
+
+			if len(tc.separators) == 0 {
+				tc.separators = append(tc.separators, "\n")
+			}
+
+			err := writeMountsFile(GetEntries(tc.nEntries, tc.nErrored, tc.nValues, tc.nDuplicated, tc.separators), WithMountsFilePath(mountsPath))
```

Comment:
I would keep GetEntries in a separate line, that way, you have:

- Setup, with clear inputs you are preparing
- The call itself, which is what you want to test
- Then the assertions.